codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -434,9 +453,10 @@ public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data) throws Http2E
         @Override
         public void onPushPromiseRead(ChannelHandlerContext ctx, int streamId, int promisedStreamId,
                 Http2Headers headers, int padding) throws Http2Exception {
-            Http2Stream parentStream = connection.requireStream(streamId);
+            Http2Stream parentStream = connection.stream(streamId);
 
-            if (streamCreatedAfterGoAwaySent(parentStream)) {
+            if (parentStream == null || streamCreatedAfterGoAwaySent(streamId)) {
```

Comment:
I think this is an error in this case [section6.6](https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-6.6):

> PUSH_PROMISE frames MUST be associated with a peer-initiated stream
   that is in either the "open" or "half closed (remote)" state.  The
   stream identifier of a PUSH_PROMISE frame indicates the stream it is
   associated with.  If the stream identifier field specifies the value
   0x0, a recipient MUST respond with a connection error (Section 5.4.1)
   of type PROTOCOL_ERROR.